nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -919,7 +917,7 @@ private boolean fieldAlwaysInitializedBeforeRead(
       if (classTreePath == null) {
         throw new IllegalStateException(
             "could not find enclosing class / enum / interface for "
-                + enclosingBlockPath.getLeaf());
+                + state.getSourceForNode(enclosingBlockPath.getLeaf()));
```

Comment:
Minor concern here: `state. getSourceForNode` performs file IO to retrieve the code, where `JTree.toString()` uses a visitor to reconstruct the string representation from the in-memory AST. This means `toString()` is faster. As long as we are only calling `getSourceForNode` for reporting actual errors (with imply IO anyways), it's probably fine to call this API, but we don't want to use it anywhere else, as it might cause a performance regression.

Related: `getSourceForNode` uses `CompilationUnitTree.getSourceFile`, which just reads `JCCompilationUnit.sourcefile` in JavaC. Do we know if that field is always set? Can javac operate on streams of java sources that are not files? (The JVM certainly can when loading bytecode, so... )

Either way, I think this is all fine for our use cases until we see an issue from this (either performance related or with getting the file) but well... any thoughts, @msridhar ?

Reply:
This is fine I think.  We should not be turning ASTs into Strings in any perf-critical computation.

Reply:
Yep agreed. Double checked and this is only used in exception messaging